contracts/light-clients/AttestorLightClient.sol

```diff
@@ -0,0 +1,216 @@
+            _grantRole(PROOF_SUBMITTER_ROLE, address(0)); // allow anyone
+        } else {
+            _grantRole(DEFAULT_ADMIN_ROLE, roleManager);
+            _grantRole(PROOF_SUBMITTER_ROLE, roleManager);
+        }
+    }
+
+    /// @inheritdoc ILightClient
+    function getClientState() external view returns (bytes memory) {
+        return abi.encode(clientState);
+    }
+
+    /// @inheritdoc IAttestorLightClient
+    function getAttestorSet() external view returns (address[] memory attestorAddresses, uint8 minRequiredSigs) {
+        return (clientState.attestorAddresses, clientState.minRequiredSigs);
+    }
+
+    /// @inheritdoc IAttestorLightClient
+    function getConsensusTimestamp(uint64 revisionHeight) external view returns (uint64) {
+        return _consensusTimestampAtHeight[revisionHeight];
+    }
+
+    /// @inheritdoc ILightClient
+    function updateClient(bytes calldata updateMsg)
+        external
+        notFrozen
+        onlyProofSubmitter
+        returns (ILightClientMsgs.UpdateResult)
+    {
+        IAttestorMsgs.AttestationProof memory proof = abi.decode(updateMsg, (IAttestorMsgs.AttestationProof));
+
+        bytes32 digest = sha256(proof.attestationData);
+        _verifySignatures(digest, proof.signatures);
+
+        IAttestorMsgs.StateAttestation memory state =
+            abi.decode(proof.attestationData, (IAttestorMsgs.StateAttestation));
+
+        // Check if height already exists, if it does, check if the timestamp is the same, otherwise revert
+        if (_consensusTimestampAtHeight[state.height] != 0) {
+            if (_consensusTimestampAtHeight[state.height] != state.timestamp) {
+                revert ConflictingTimestamp(state.height, _consensusTimestampAtHeight[state.height], state.timestamp);
+            }
+            return ILightClientMsgs.UpdateResult.NoOp;
+        }
+
+        _consensusTimestampAtHeight[state.height] = state.timestamp;
+        clientState.latestHeight = state.height;
+        return ILightClientMsgs.UpdateResult.Update;
+    }
+
+    /// @inheritdoc ILightClient
+    function verifyMembership(ILightClientMsgs.MsgVerifyMembership calldata msg_)
+        external
+        view
+        notFrozen
+        onlyProofSubmitter
+        returns (uint256)
+    {
+        if (msg_.value.length == 0) revert EmptyValue();
+
+        // Ensure we have a trusted timestamp at the provided height.
+        uint64 height = msg_.proofHeight.revisionHeight;
+        uint64 ts = _consensusTimestampAtHeight[height];
+        if (ts == 0) revert ConsensusTimestampNotFound(height);
+
+        IAttestorMsgs.AttestationProof memory proof = abi.decode(msg_.proof, (IAttestorMsgs.AttestationProof));
+        bytes32 digest = sha256(proof.attestationData);
+        _verifySignatures(digest, proof.signatures);
+
+        // Decode the attested packet commitments and verify the attested height matches the provided proof height
+        IAttestorMsgs.PacketAttestation memory packetAttestation =
```

Comment:
Might be good to require that `packetAttestation.packets.length != 0`. Although this doesn't pose an immediate security risk